rfc/multiple-go-modules.md

```diff
@@ -0,0 +1,188 @@
+| ------------ | -------------------------------- | ----------------------------- |
+| provider     | example_external_provider_plugin | provider/examples/external    |
+| cli          | scanner_boot_test                | cli/test/boot                 |
+|              | img                              | assets                        |
+
+Makefile, GitHub workflows and other files will need to be updated to point to the new paths.
+
+The VMClarity directory will have the following structure.
+
+```sh
+.
+├── Makefile
+├── api
+│   ├── client
+│   │   ├── client.cfg.yaml
+│   │   └── go.mod
+│   ├── go.mod
+│   ├── models
+│   │   └── models.cfg.yaml
+│   ├── openapi.yaml
+│   └── server
+│       ├── cmd
+│       ├── go.mod
+│       ├── pkg
+│       │   ├── common
+│       │   ├── database
+│       │   └── rest
+│       └── server.cfg.yaml
+├── cli
+│   ├── cmd
+│   ├── go.mod
+│   ├── pkg
+│   │   ├── analyzer
+│   │   ├── config
+│   │   ├── converter
+│   │   ├── families
+│   │   ├── findingkey
+│   │   ├── jobmanager
+│   │   ├── presenter
+│   │   ├── scanner
+│   │   ├── state
+│   │   └── utils
+│   └── test
+│       └── boot
+├── docs
+├── e2e
+│   ├── config
+│   ├── go.mod
+│   └── testenv
+│       ├── docker
+│       ├── kubernetes
+│       ├── types
+│       └── utils
+├── assets
+├── installation
+│   ├── aws
+│   ├── azure
+│   ├── docker
+│   ├── gcp
+│   └── kubernetes
+├── orchestrator
+│   ├── cmd
+│   ├── go.mod
+│   └── pkg
+│       ├── assetscanestimationwatcher
+│       ├── assetscanprocessor
+│       ├── assetscanwatcher
+│       ├── common
+│       ├── discovery
+│       ├── scanconfigwatcher
+│       ├── scanestimationwatcher
+│       └── scanwatcher
+├── provider
+│   ├── cmd
+│   ├── examples
+│   │   └── external
+│   ├── go.mod
+│   └── pkg
+│       ├── aws
+│       ├── azure
+│       ├── cloudinit
+│       ├── common
+│           └── containerruntimediscovery
+│               └── cmd
+│       ├── docker
+│       ├── external
+│       ├── gcp
+│       └── kubernetes
+├── rfc
+├── ui
+│   └── src
+├── uibackend
+│   ├── client
+│   │   └── client.cfg.yaml
+│   ├── go.mod
+│   ├── models
+│   │   └── models.cfg.yaml
+│   ├── openapi.yaml
+│   └── server
+│       ├── cmd
+│       ├── go.mod
+│       ├── pkg
+│       └── server.cfg.yaml
+└── utils
+    ├── command
+    ├── fsutils
+    ├── go.mod
+    ├── log
+    ├── manifest
+    └── version
+```
+
+# Release
+
+Each module will have a tag with the format `prefix/version` where prefix is the directory within the repository where the module is defined, more details [here](https://go.dev/wiki/Modules#publishing-a-release). For now, the same version will be used for each module even if there are no changes, this will simplify managing compatibility between modules across versions.
+
+The release GitHub workflow will be extended with an additional step to create a tag for each module. This step will only be performed after the release is successfully published.
```

Comment:
I think we need to include a plan for the release process as there are a couple of open questions still:

Releasing all the modules in one go would work using `replace` directives for local modules packages in `go.mod` for modules in the repository. However the release process also needs to ensure that the module versions are bumped in `go.mod` as `replace` directives are not considered when a modules is being imported by 3rd party.

Reply:
just pushed a commit extending the release process changes, let me know what you think 😃